sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1192,10 +1193,18 @@ class Analyzer(
    * @see https://issues.apache.org/jira/browse/SPARK-19737
    */
   object LookupFunctions extends Rule[LogicalPlan] {
-    override def apply(plan: LogicalPlan): LogicalPlan = plan.transformAllExpressions {
-      case f: UnresolvedFunction if !catalog.functionExists(f.name) =>
-        withPosition(f) {
-          throw new NoSuchFunctionException(f.name.database.getOrElse("default"), f.name.funcName)
+    override def apply(plan: LogicalPlan): LogicalPlan = {
+      val catalogFunctionNameSet = new mutable.HashSet[FunctionIdentifier]()
+        plan.transformAllExpressions {
+          case f: UnresolvedFunction if catalogFunctionNameSet.contains(f.name) => f
+          case f: UnresolvedFunction if catalog.functionExists(f.name) =>
+            catalogFunctionNameSet.add(f.name)
```

Comment:
Normalize the name before adding it to the cache? This can cover more cases

Reply:
sure, I will do that. Thanks.